javascript/ql/lib/semmle/javascript/internal/flow_summaries/Arrays.qll

```diff
@@ -486,6 +486,11 @@ class Shift extends SummarizedCallable {
     preservesValue = true and
     input = "Argument[this].ArrayElement[0]" and
     output = "ReturnValue"
+    or
+    // ArrayElement[0] is not automatically converted to a taint step, so add it manually
```

Comment:
I don't think this comment is placed at the right place?
Or you meant `Argument[this]` instead of `ArrayElement[0]`?

Reply:
It refers to the `ArrayElement[0]` in the summary above here.

`ArrayElement` is automatically converted to a taint step, but `ArrayElement[0]` is not, which is why we need this extra summary.

Reply:
Oh, now I get it. 
Maybe a reword like:  
`// The above doesn't add a taint-step from the array itself to the return value, so we add that manually.`